modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -843,6 +843,321 @@ static void run_arithm_s(DST out[], const SRC in[], int width, int chan,
+        }
+
+        if (x < length && (in != out))
+        {
+            x = length - 3 * nlanes;
+            continue;  // process one more time (unaligned tail)
+        }
+        break;
+    }
+    return x;
+}
+
+template<>
+CV_ALWAYS_INLINE int absdiffc_simd_c3_impl<uchar>(const uchar in[], uchar out[],
+                                                  const v_float32& s1, const v_float32& s2,
+                                                  const v_float32& s3, const int length)
+{
+    constexpr int nlanes = static_cast<int>(v_uint8::nlanes);
+    constexpr int num_vectors = 12;
+
+    if (length < 3 * nlanes)
+        return 0;
+
+    int x = 0;
+
+    v_float32 vectors[num_vectors];
+
+    for (;;)
+    {
+        for (; x <= length - 3 * nlanes; x += 3 * nlanes)
+        {
+            for (int i = 0; i < num_vectors; ++i)
+            {
+                vectors[i] = v_cvt_f32(v_reinterpret_as_s32(vx_load_expand_q(in + x + i * nlanes / 4)));
+            }
+
+            vx_store(&out[x], v_pack_u(v_pack(v_round(v_absdiff(vectors[0], s1)),
+                                              v_round(v_absdiff(vectors[1], s2))),
+                                       v_pack(v_round(v_absdiff(vectors[2], s3)),
+                                              v_round(v_absdiff(vectors[3], s1)))));
+
+            vx_store(&out[x + nlanes], v_pack_u(v_pack(v_round(v_absdiff(vectors[4], s2)),
+                                                       v_round(v_absdiff(vectors[5], s3))),
+                                                v_pack(v_round(v_absdiff(vectors[6], s1)),
+                                                       v_round(v_absdiff(vectors[7], s2)))));
+
+            vx_store(&out[x + 2 * nlanes], v_pack_u(v_pack(v_round(v_absdiff(vectors[8], s3)),
+                                                           v_round(v_absdiff(vectors[9], s1))),
+                                                    v_pack(v_round(v_absdiff(vectors[10], s2)),
+                                                           v_round(v_absdiff(vectors[11], s3)))));
+        }
+
+        if (x < length && (in != out))
+        {
+            x = length - 3 * nlanes;
+            continue;  // process one more time (unaligned tail)
+        }
+        break;
+    }
+    return x;
+}
+
+template<typename T>
+CV_ALWAYS_INLINE int absdiffc_simd_c3(const T in[], const float scalar[], T out[], int width)
+{
+    constexpr int chan = 3;
+    constexpr int size = static_cast<int>(v_float32::nlanes) + 2;
+    int length = width * chan;
+
+    float init[size];
+    for (int i = 0; i < size; ++i)
+    {
+        init[i] = *(scalar + i % chan);
+    }
+
+    v_float32 s1 = vx_load(init);
+
+#if (CV_AVX512_SKX | (CV_SSE2 && !CV_AVX2) | CV_NEON)
+    v_float32 s2 = vx_load(init + 1);
+    v_float32 s3 = vx_load(init + 2);
+#elif CV_AVX2
+    v_float32 s2 = vx_load(init + 2);
+    v_float32 s3 = vx_load(init + 1);
+#endif
```

Comment:
Why do we need such difference between native instructions?

Where is `#else` branch implementation?

Reply:
@alalek  The difference is related to the lengths of SIMD vectors. For example let's compare SSE42 and AVX2. This function handle 3-channel case. In current snippet I load float coefficients. 

SSE42 and NEON cases:
A vector have 128 bits. So I can load to it 4 32-bit float numbers. So after loading the first,  the second and third vectors look like these:

First:  C1 C2 C3 _**C1**_
Second:   _**C2**_ C3 C1 _**C2**_
Third:   _**C3**_ C1 C2 C3

As you remember I load vectors from `init[]` array that look like this ` float init[ nlanes+2 ] = {C1, C2, C3, C1, C2, C3 ...};`

Therefore, in all cases **_except AVX2_**, the second vector starts loading since second `init[]` element (init[1] = C2) and third vector starts loading at third `init[]` element (init[2] = C3).

AVX2 case:
A vector have 256 bits. So I can load to it 8 32-bit floats. So 3 vectors look like these:

First: C1 C2 C3 C1 C2 C3 C1 _**C2**_
Second: _**C3**_ C1 C2 C3 C1 C2 C3 _**C1**_
Third:  **_C2_** C3 C1 C2 C3 C1 C2 C3

So, in AVX2 case the second vector starts loading since third `init[]` element (init[2] = C3) and third vector starts loading at second `init[]` element (init[1] = C2).

So, It is the reason for the difference in this snippet.

Reply:
> Where is `#else` branch implementation?

There is no need in `#else` branch because all this code is under CV_SIMD macro and all SIMD cases handles by 2 these branches mentioned above.

Reply:
> this function is under CV_SIMD macro and all SIMD cases handles by 2 current branches mentioned above

CV_SIMD includes these cases too: CV_VSX, CV_VSX3, CV_MSA, CV_WASM_SIMD, CV_RVV

Without "#else" you **silently** ignore error cases.

---

> A vector have 256 bits

So, you need to check for `CV_SIMD_WIDTH` value instead:
- CV_SIMD_WIDTH = 16 (simd128)
- CV_SIMD_WIDTH = 32 (simd256)
- CV_SIMD_WIDTH = 64 (simd512)

Reply:
Applied.